### PR TITLE
[WIP] Migration to python3

### PR DESCRIPTION
**Pull request by wo0dyn**
# Migration to python3
- [x] Fix tests
- [x] Add `py.test` to run tests locally
- [x] Add tests for `CachableModel`
- [x] Add `SlugableModel`
- [ ] Refactor tests
- [ ] Write a tiny doc
- [ ] Bump a new version (maybe 1.0.0)

### CODE REVIEW

**Comment by wo0dyn** [comment]
Yeah, looks nice @harmo! :wink: Thanks.

File: README.rst [hunk 1]
```diff
@@ -84,9 +97,10 @@ Tests
...
 
-    $ cd tests && python manage.py test
+    $ pip install -r tests/requirements.txt
+    $ py.test --ds=tests.settings tests
 
 Authors
 -------
 
-Nicolas Dubois <nicolas.c.dubois@gmail.com>
+@wo0dyn • Nicolas Dubois <nicolas.c.dubois@gmail.com>
```
Comment (wo0dyn) [inline README.rst:106]: @harmo › Add your name right there :wink:

File: basic_models_behaviors/managers.py [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
+
+    def get_queryset(self):
+        return CachedQuerySet(self.model)
+
+
+def get_key_for_instance(instance, pk=None):
+    return '{label}.{module}:{pk}'.format(
+        label=instance._meta.app_label,
+        module=instance._meta.model_name,
+        pk=instance.pk if pk is None else pk)
```
Comment (wo0dyn) [inline basic_models_behaviors/managers.py:49]: :+1:

File: basic_models_behaviors/models.py [hunk 3]
```diff
@@ -3,7 +3,10 @@
 from datetime import datetime
 
 from django.db import models
-from managers import CacheManager
+from django.core.cache import cache
+
+from .managers import CacheManager
+from basic_models_behaviors.managers import get_key_for_instance
```
Comment (wo0dyn) [inline basic_models_behaviors/models.py]:
``` python
from .managers import CacheManager, get_key_for_instance
```

Comment (harmo) [inline basic_models_behaviors/models.py]: Oops

**Comment by harmo** [comment]
@wo0dyn 
The third point is not finished yet, there missing a lot of tests on multiples objects returned during a filter, for example.

**Comment by wo0dyn** [comment]
@harmo • Note: I re-wrote the history a little bit (squashing some commits that were actually related, rewording some commit messages to fix typos).

File: README.rst (hunk 1 above)
Comment (wo0dyn) [inline README.rst:106]: Done. I put your GitHub username, instead of your Twitter's.

File: tests/test_models.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+
+class SlugableModelTests(TestCase):
+
+    def setUp(self):
+        self.sm = models.SlugableMock.objects.create(
+            label='Never play leapfrog on a unicorn!')
+
+    def test_simple_sluggable_model_creation(self):
+        self.assertEquals(self.sm.slug, 'never-play-leapfrog-on-a-unicorn')
```
Comment (wo0dyn) [inline tests/test_models.py:162]: Hey @harmo, I believe that the slug should have been `never-play-leapfrog-on-a-unicorn` for the object `SlugableMock.objects.create(label='Never play leapfrog on a unicorn!')`. Instead we have: `slugablemock-object`.

What do you think?

Comment (harmo) [inline tests/test_models.py:162]: That's just because there is no `__unicode__`, but you are right.

**Comment by wo0dyn** [comment]
Stale PR.

**Pull request closed without merging**